### PR TITLE
IABot: Acceptance tests!

### PR DESCRIPTION
**Pull request by MusikAnimal**
Covers most bug reports from July to September 2016

Bug: T140386

### CODE REVIEW

File: IABot/deadlink.php [hunk 1]
```diff
@@ -222,5 +222,7 @@ function( $thread ) {
 	if( DEBUG === false && LIMITEDRUN === false ) DB::generateLogReport();
 	if( file_exists( IAPROGRESS.WIKIPEDIA.UNIQUEID."stats" ) && LIMITEDRUN === false ) unlink( IAPROGRESS.WIKIPEDIA.UNIQUEID."stats" );
 	if( DEBUG === false && LIMITEDRUN === false ) sleep(10);
-	if( DEBUG === true || LIMITEDRUN === true ) exit(0);
+
+	// return instead of exiting so that acceptance tests will finish
+	if( DEBUG === true || LIMITEDRUN === true ) return;
```
Comment (cyberpower678) [inline IABot/deadlink.php]: Return from what?

File: IABot/tests/acceptance/test_cases/acceptance-result.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Some text.
+<ref name="raw-live-url">https://en.wikipedia.org/wiki/Main_Page</ref>
+<ref name="raw-dead-url">{{cite web|url=http://www.statistics.sk/mosmis/eng/run.html |title=Archived copy |accessdate=2007-05-13 |deadurl=yes |archiveurl=https://web.archive.org/web/20070513023228/http://www.statistics.sk/mosmis/eng/run.html |archivedate=2007-05-28 |df= }}</ref>
+<ref name="raw-dead-url-w-text">Doe, Jane (1977). http://www.statistics.sk/mosmis/eng/run.html.{{dead link}} More text.</ref>
```
Comment (cyberpower678) [inline IABot/tests/acceptance/test_cases/acceptance-result.txt]: Shouldn't an archive be here if it's dead?

Comment (cyberpower678) [inline IABot/tests/acceptance/test_cases/acceptance-result.txt]: Oh right, that period.  It stops grabbing the contents of a plain URL if it's not a whitespace.

File: IABot/tests/acceptance/test_cases/acceptance-result.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<ref name="raw-dead-link">[http://www.statistics.sk/mosmis/eng/run.html "Statistics"] {{wayback|url=http://www.statistics.sk/mosmis/eng/run.html |date=20070528023228 }}</ref>
+<ref name="raw-dead-link-w-text">Doe, Jane (1977). [http://www.statistics.sk/mosmis/eng/run.html "Statistics"] {{wayback|url=http://www.statistics.sk/mosmis/eng/run.html |date=20070528023228 }}. More text.</ref>
+<ref name="raw-dead-link-w-archive">[http://www.statistics.sk/mosmis/eng/run.html "Statistics"]. ([https://web.archive.org/web/20070513023228/http://www.statistics.sk/mosmis/eng/run.html Archive])</ref>
+<ref name="raw-dead-url-and-live-url">http://www.statistics.sk/mosmis/eng/run.html.{{dead link}} https://en.wikipedia.org/wiki/Main_Page.</ref>
```
Comment (cyberpower678) [inline IABot/tests/acceptance/test_cases/acceptance-result.txt]: dead tags should really be attached to the end of the URL itself not, after the end of a sentence.

File: IABot/deadlink.config.inc.php [hunk 4]
```diff
@@ -79,7 +79,7 @@
 	if( $multithread || $workers ) require_once( 'thread.php' );
 	require_once( 'Parser/parse.php' );
 	require_once( 'DB.php' );
-	require_once( '../vendor/autoload.php' );
+	require_once( dirname( __DIR__ ) . '/vendor/autoload.php' );
```
Comment (kaldari) [inline IABot/deadlink.config.inc.php]: This should just be __DIR__, not dirname( __DIR__ ).

Comment (cyberpower678) [inline IABot/deadlink.config.inc.php]: This is currently fixed in the deployed version.

**Pull request merged**